routing/routing_algorithm.cpp

```diff
@@ -19,7 +19,14 @@ double constexpr KMPH2MPS = 1000.0 / (60 * 60);
 inline double TimeBetweenSec(Junction const & j1, Junction const & j2, double speedMPS)
 {
   ASSERT(speedMPS > 0.0, ());
-  return MercatorBounds::DistanceOnEarth(j1.GetPoint(), j2.GetPoint()) / speedMPS;
+  double const distanceM = MercatorBounds::DistanceOnEarth(j1.GetPoint(), j2.GetPoint());
+  feature::TAltitude const j1Altitude = j1.GetAltitude();
+  feature::TAltitude const j2Altitude = j2.GetAltitude();
+  if (j1Altitude == feature::kInvalidAltitude || j2Altitude == feature::kInvalidAltitude)
```

Comment:
For metric to be consistent, you need to treat to use min mwm altitude instead of kInvalidAltitude.